sim/src/cap.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{CarID, VehicleType};
+use geom::{Duration, Time};
+use map_model::{LaneID, Map, Path, PathConstraints, PathStep};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+// Note this only indexes into the zones we track here, not all of them in the map.
+type ZoneIdx = usize;
+
+// This only caps driving trips.
+#[derive(Serialize, Deserialize, PartialEq, Clone)]
+pub struct CapSimState {
+    lane_to_zone: BTreeMap<LaneID, ZoneIdx>,
+    zones: Vec<Zone>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Clone)]
+struct Zone {
+    cap: usize,
+    entered_in_last_hour: BTreeSet<CarID>,
+    // TODO Maybe want sliding windows or something else
```

Comment:
There are a bunch of ways to express a cap of vehicles passing through some zone. Just doing the really dumb thing of 1-2am, 2-3am, etc until the other folks are more specific.